hamburgerMenu.js

```diff
@@ -1,19 +1,28 @@
-const navHamburger = document.getElementById("nav__hamburger");
 const hamburgerMenu = document.getElementById("nav__hamburger-menu");
 const hamburgerIcon = document.getElementById("nav__hamburger-icon");
 
-hamburgerIcon.addEventListener("click", toggleMenu);
-
-function toggleMenu () {
-  if(!cartMenu.classList.contains("--off")) {
-    cartMenu.classList.toggle("--off");
-  }
-  if (!hamburgerMenu.classList.contains("--off")) {
-    hamburgerMenu.classList.toggle("--off");
-    hamburgerIcon.classList.toggle("change");
+document.addEventListener("click", function(event){
+	if (event.target==hamburgerIcon) {
+        if(!cartMenu.classList.contains("--hidden")) {
+            cartMenu.classList.toggle("--hidden");
+        }
+        if (!hamburgerMenu.classList.contains("--hidden")) {
+            hamburgerMenu.classList.toggle("--hidden");
+            hamburgerIcon.classList.toggle("change");
+        }
+        else{
+            hamburgerMenu.classList.toggle("--hidden");
+            hamburgerIcon.classList.toggle("change");
```

Comment:
What is the difference between the block body `if` and `else`?

Reply:
It's a comment just note for thinking, required actions described in the algorithm for optimization